src/gurobi_optimods/min_cut.py

```diff
@@ -0,0 +1,289 @@
+
+    Returns
+    -------
+    cut: float
+        Cut value of the minimum cut.
+    partition: tuple[set[int], set[int]]
+        Partition of size 2 with cut sets.
+    cutset: set[tuple[int, int]]
+        Cutset with edges.
+    """
+    if isinstance(graph, sp.spmatrix):
+        return _min_cut_scipy(graph, source, sink, create_env)
+    elif isinstance(graph, pd.DataFrame):
+        return _min_cut_pandas(graph, source, sink, create_env)
+    elif nx is not None and isinstance(graph, nx.Graph):
+        return _min_cut_networkx(graph, source, sink, create_env)
+    else:
+        raise ValueError(f"Unknown graph type: {type(graph)}")
+
+
+def _min_cut_pandas(arc_data, source, sink, create_env):
+    f, t = arc_data.index.names
+    arc_data["cost"] = [0] * len(arc_data)
+    # Create dummy edge to find maximum flow through (minimum of sum of all
+    # outgoing/incoming capacities at the source/sink)
+    max_flow = min(
+        arc_data.loc[([source], slice(None)),]["capacity"].sum(),
+        arc_data.loc[(slice(None), [sink]),]["capacity"].sum(),
+    )
+    arc_data = pd.concat(
+        [
+            arc_data,
+            pd.DataFrame(
+                [{f: sink, t: source, "capacity": max_flow, "cost": 1}]
+            ).set_index([f, t]),
+        ]
+    )
+
+    with create_env() as env, gp.Model(env=env) as model:
+        # Solve max-flow problem
+        model.ModelSense = GRB.MAXIMIZE
+        arc_df = arc_data.gppd.add_vars(model, obj="cost", name="flow")
+        balance_df = pd.DataFrame(
+            {
+                "inflow": arc_df["flow"].groupby(t).sum(),
+                "outflow": arc_df["flow"].groupby(f).sum(),
+            }
+        ).gppd.add_constrs(model, "inflow == outflow", name="balance")
+        capacity_constrs = gppd.add_constrs(
+            model,
+            arc_df["flow"],
+            GRB.LESS_EQUAL,
+            arc_df["capacity"],
+            name="capacity",
+        )
+        logger.info(
+            f"Solving min-cut problem with {len(balance_df)} nodes and "
+            f"{len(arc_data)-1} edges"
+        )
+        model.optimize()
+
+        if model.Status in [GRB.INFEASIBLE, GRB.INF_OR_UNBD]:
+            raise ValueError("Unsatisfiable flows")
+
+        # Construct partition and cutset
+
+        cap_pi = capacity_constrs.gppd.Pi
+
+        cutset = set([a for a in arc_data.index if cap_pi[a] > 1e-3])
+
+        p1 = set()
+        queue = [source]
+        while len(queue) > 0:
+            node = queue.pop()
+            p1.add(node)
+            # Add successors of `node` that are not in the cutset
+            queue.extend(
+                [
+                    a[1]
+                    for a in arc_data.loc[([node], slice(None)),].index
+                    if a not in cutset and a[1] not in p1 and a[1] not in queue
+                ]
+            )
+        p2 = set([n for n in balance_df.index if n not in p1])
+        return model.ObjVal, (p1, p2), cutset
```

Comment:
I think this is too many values to return as a nested tuple. Maybe the result could be wrapped up in a dataclass which is separately documented?